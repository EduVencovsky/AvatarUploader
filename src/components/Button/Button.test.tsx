import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { Button } from '.';

it('renders correctly', async () => {
  const { queryByTestId } = render(<Button>content</Button>);
  expect(queryByTestId('styled-button')).toMatchSnapshot();
});

it('renders snapshot', async () => {
  const { container } = render(<Button textColor="red">click me</Button>);
  expect(container).toMatchSnapshot();
});

it('triggers clicks correctly', async () => {
  const onClick = jest.fn()
  const { getByTestId } = render(<Button onClick={onClick}>click me</Button>);
  fireEvent.click(getByTestId('styled-button'))
  expect(onClick).toBeCalledTimes(1);
});
