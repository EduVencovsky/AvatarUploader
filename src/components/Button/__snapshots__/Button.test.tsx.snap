// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.emotion-0 {
  color: primary;
  background: none;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  outline: inherit;
}

<button
  class="emotion-0"
  data-testid="styled-button"
>
  content
</button>
`;

exports[`renders snapshot 1`] = `
.emotion-0 {
  color: primary;
  background: none;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  outline: inherit;
}

<div>
  <button
    class="emotion-0"
    data-testid="styled-button"
  >
    click me
  </button>
</div>
`;
